eap-rhel-payg/src/main/bicep/mainTemplate.bicep

```diff
@@ -304,7 +304,7 @@ resource vmName_resource 'Microsoft.Compute/virtualMachines@${azure.apiVersionFo
         publisher: 'RedHat'
         offer: 'rh-jboss-eap'
         sku: (jdkVersion == 'eap74-openjdk8') ? 'rh-jboss-eap74-jdk8-rhel8' : (jdkVersion == 'eap74-openjdk11') ? 'rh-jboss-eap74-jdk11-rhel8' : (jdkVersion == 'eap74-openjdk17') ? 'rh-jboss-eap74-jdk17-rhel8' :  (jdkVersion == 'eap8-openjdk11') ? 'rh-jboss-eap8-jdk11-rhel9' :  (jdkVersion == 'eap8-openjdk17') ? 'rh-jboss-eap8-jdk17-rhel9' :  null
-        version: (jdkVersion == 'eap8-openjdk11') ? '8.0.2024111801' :  (jdkVersion == 'eap8-openjdk17') ? '8.0.2024111801' :  'latest'
+        version: 'latest'
```

Comment:
Can you also switch skus for eap8 as well?

```
- .... (jdkVersion == 'eap8-openjdk11') ? 'rh-jboss-eap8-jdk11-rhel9' : (jdkVersion == 'eap8-openjdk17') ? 'rh-jboss-eap8-jdk17-rhel9' : null
+ .... (jdkVersion == 'eap8-openjdk11') ? 'rh-jboss-eap8-jdk11-rhel9-gen2' : (jdkVersion == 'eap8-openjdk17') ? 'rh-jboss-eap8-jdk17-rhel9-gen2' : null

```

It will be great to always use `hyperVGeneration = "V2"`. Feel free to do it in different PRs

Reply:
Sure will update the EAP 8 ones to use gen2.

Reply:
I think by default it was picking gen2. I think we tested this out.
Not for EAP 8
```
sh$ for i in redhat:rh-jboss-eap:rh-jboss-eap8-jdk1{1,7}-rhel9:latest; do echo "*** urn: $i"; az vm image show --urn "$i" | jq '.name + " gen:" +.hyperVGeneration, .imageDeprecationStatus' ; done
*** urn: redhat:rh-jboss-eap:rh-jboss-eap8-jdk11-rhel9:latest
"8.0.2024122012 gen:V1"
{
  "imageState": "Active"
}
*** urn: redhat:rh-jboss-eap:rh-jboss-eap8-jdk17-rhel9:latest
"8.0.2024122012 gen:V1"
{
  "imageState": "Active"
}
```

Reply:
ok thanks, Updated!